From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Armbian Build <info@armbian.com>
Date: Wed, 04 Dec 2024 01:30:00 +0000
Subject: Add RK3128-box device tree for U-Boot

Add device tree support for generic RK3128-based TV boxes with:
- eMMC and SD card boot support
- Serial console
- Minimal configuration for bootloader

Based on rk3128-evb and adapted for TV box hardware.

---
 arch/arm/dts/rk3128-box.dts         | 53 +++++++++++++++++++++++++++++++
 arch/arm/dts/rk3128-box-u-boot.dtsi |  3 ++
 2 files changed, 56 insertions(+)

diff --git a/arch/arm/dts/rk3128-box.dts b/arch/arm/dts/rk3128-box.dts
new file mode 100644
index 000000000000..111111111111
--- /dev/null
+++ b/arch/arm/dts/rk3128-box.dts
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: GPL-2.0+ OR X11
+/*
+ * (C) Copyright 2024 Armbian Build
+ *
+ * Generic rk3128 tv box device tree for U-Boot.
+ */
+
+/dts-v1/;
+
+#include "rk3128.dtsi"
+
+/ {
+	model = "Generic Rockchip RK3128 TV Box";
+	compatible = "rockchip,rk3128-box", "rockchip,rk3128";
+
+	chosen {
+		bootph-all;
+		stdout-path = &uart2;
+		u-boot,spl-boot-order = "same-as-spl", &sdmmc, &emmc;
+	};
+
+	memory@60000000 {
+		device_type = "memory";
+		reg = <0x60000000 0x40000000>;
+	};
+
+	vcc_sys: vcc-sys-regulator {
+		compatible = "regulator-fixed";
+		regulator-name = "vcc_sys";
+		regulator-always-on;
+		regulator-boot-on;
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+	};
+};
+
+&emmc {
+	bootph-pre-ram;
+	bus-width = <8>;
+	non-removable;
+	status = "okay";
+};
+
+&sdmmc {
+	bootph-pre-ram;
+	bus-width = <4>;
+	status = "okay";
+};
+
+&uart2 {
+	bootph-all;
+	status = "okay";
+};
diff --git a/arch/arm/dts/rk3128-box-u-boot.dtsi b/arch/arm/dts/rk3128-box-u-boot.dtsi
new file mode 100644
index 000000000000..222222222222
--- /dev/null
+++ b/arch/arm/dts/rk3128-box-u-boot.dtsi
@@ -0,0 +1,3 @@
+// SPDX-License-Identifier: GPL-2.0+ OR X11
+
+#include "rk3128-u-boot.dtsi"
--
Armbian
